app/models/supplejack_api/top_collection_metric.rb

```diff
@@ -0,0 +1,136 @@
+
+            # If there are no results for a metric, date, and display collection
+            # Skip to the next metric
+            next if results.select { |_key, value| value.positive? }.empty?
+
+            top_collection_metric = find_or_create_top_collection_metric(date, metric, dc)
+            update_top_collection_metric(top_collection_metric, results)
+            metrics.push(top_collection_metric)
+          end
+        end
+        stamp_record_metrics(date)
+      end
+
+      disable_metrics_logger
+
+      metrics
+    end
+
+    def self.find_dates_to_process
+      METRICS_LOGGER.info 'Collecting Dates.'
+
+      query = { :date.lt => Time.zone.now.beginning_of_day }
+      SupplejackApi::RecordMetric.where(query)
+                                 .map(&:date).uniq
+    end
+
+    def self.find_display_collections_to_process
+      METRICS_LOGGER.info 'Collecting Collections.'
+
+      query = { :date.lt => Time.zone.now.beginning_of_day, processed_by_top_collection_metrics: false }
+      SupplejackApi::RecordMetric.where(query)
+                                 .map(&:display_collection).uniq.compact
+    end
+
+    def self.calculate_results(record_metrics, metric)
+      record_metrics.each_with_object({}) do |record, hash|
+        hash[record.record_id.to_s] = record.send(metric)
+      end
+    end
+
+    def self.update_top_collection_metric(top_collection_metric, results)
+      if top_collection_metric.results.blank?
+        top_collection_metric.update(results: results)
+      else
+        merged_results = top_collection_metric.results.merge(results) { |_key, a, b| a + b }
+        merged_results = merged_results.sort_by { |_k, v| -v }.first(200).to_h
+
+        top_collection_metric.update(results: merged_results)
+      end
+
+      METRICS_LOGGER.info "Updated Top Collection Metric: #{top_collection_metric.inspect}"
+      top_collection_metric
+    end
+
+    def self.find_or_create_top_collection_metric(date, metric, display_collection)
+      top_collection_metric = find_or_create_by(
+        date: date,
+        metric: metric,
+        display_collection: display_collection
+      )
+
+      METRICS_LOGGER.info "Top Collection Metric: #{top_collection_metric.inspect}"
+      top_collection_metric
+    end
+
+    def self.record_metrics_to_be_processed(date, metric, display_collection)
+      SupplejackApi::RecordMetric.where(
+        date: date,
+        display_collection: display_collection,
+        :processed_by_top_collection_metrics.in => [nil, '', false]
+      ).order_by(metric => 'desc').limit(200)
+    end
+
+    def self.stamp_record_metrics(date)
+      METRICS_LOGGER.info "Stamping Record Metrics for date: #{date}"
+      SupplejackApi::RecordMetric.where(date: date).update_all(processed_by_top_collection_metrics: true)
+    end
+
+    def self.enable_metrics_logger
+      Mongoid.logger = METRICS_LOGGER
+    end
+
+    def self.disable_metrics_logger
+      Mongoid.logger = Rails.logger
```

Comment:
Once you are happy with this it would be good to remove these loggers.

Reply:
would be nice to keep them to monitor how the metrics is working in the long run.